### PR TITLE
Adding Implementation of noex::InstanceAdmin::DeleteInstance

### PR DESCRIPTION
**Pull request by roopak-qlogic**
Fixes issue #1264

---
This change is [Reviewable](https://reviewable.io/reviews/googlecloudplatform/google-cloud-cpp/1320)

### CODE REVIEW

File: google/cloud/bigtable/internal/instance_admin.h [hunk 1]
```diff
@@ -167,6 +167,53 @@ class InstanceAdmin {
 
   void DeleteInstance(std::string const& instance_id, grpc::Status& status);
 
+  /**
+   * Makes an asynchronous request to delete the instance.
```
Comment (coryan) [inline google/cloud/bigtable/internal/instance_admin.h]: s/the instance/an instance/

File: google/cloud/bigtable/internal/instance_admin.h [hunk 2]
```diff
@@ -167,6 +167,53 @@ class InstanceAdmin {
...
+   *
+   * @param instance_id the id of the instance in the project to be deleted.
+   * @param cq the completion queue that will execute the asynchronous calls,
+   *     the application must ensure that one or more threads are blocked on
+   *     `cq.Run()`.
+   * @param callback a functor to be called when the operation completes. It
+   *     must satisfy (using C++17 types):
+   *     static_assert(std::is_invocable_v<
+   *         Functor, google::protobuf::Empty&,
```
Comment (coryan) [inline google/cloud/bigtable/internal/instance_admin.h]: Do you think it is possible to change the `Functor` requirements to be `std::is_invocable_v<Functor, grpc::Status const&>` ?  That `google::protobuf::Empty` is just a placeholder, we should be able to eliminate it.  If you cannot figure out an easy way then please open a bug to fix that later.

Comment (roopak-qlogic) [reply google/cloud/bigtable/internal/instance_admin.h]: I tried to quick or easy fix, but did not find the right way to fix, so I will open a bug for this, and in background I will try to figure out how to fix

**Comment by roopak-qlogic** [comment]
Appveyor build has failed, do not know the exact reason, may need to restart

**Comment by roopak-qlogic** [comment]
waiting this code to merge for my next PR, to avoid merge conflict.

**Pull request merged**